### PR TITLE
fix: preserve literal braces in Postman imports

### LINKED ISSUE #6324: [bug]: Postman collection import converts literal `}}` in request bodies to `>>`, corrupting JSON
### Is there an existing issue for this?

- [x] I have searched existing issues and this bug hasn't been reported yet

### Platform

Desktop App

### Browser

Chrome

### Operating System

macOS

## What happened?
When importing a Postman collection (v2.1) that contains a request body with literal `}}` characters (e.g. a stringified JSON nested inside another JSON, the importer converts the `}}` sequences into `>>`, as if they were Postman environment-variable delimiters.

The result is a syntactically invalid request body. Postman's `{{var}}` syntax requires a matching opener and closer, but Hoppscotch appears to substitute the closing `}}` unconditionally, without checking that a `{{` opened a variable reference.

Example: the source collection contains {"id":4,"required":1,"data":{"type":12,"len":32,"ext":{"required":["buttonalign","buttonstyle"],"filesize":"1MB"}}} which is a perfectly valid JSON fragment (three closing braces to close `ext`, `data`, and the asset object).
  
After import into Hoppscotch, the body contains: {"id":4,"required":1,"data":{"type":12,"len":32,"ext":{"required":["buttonalign","buttonstyle"],"filesize":"1MB">>}

The first `}}` was rewritten to `>>` (and one trailing `}` survives), producing invalid JSON that the server rejects.

Note: legitimate Postman variables in the same collection (e.g. `{{serviceInternalUrl}}` in the URL) are correctly translated to `<<serviceInternalUrl>>`, so the variable-substitution mechanism is working — it's just over-eager and matches `}}` (and maybe `{{`?) anywhere in any string field, not only inside paired `{{...}}` references.

## Steps to reproduce
1. In Hoppscotch, go to Collections → Import/Export → Import from Postman → upload attached `repro.json`.
2. Observe that the raw body contains `>>` where the original had `}}`, and that the body is no longer valid JSON.
  
Expected: after import, the body should be byte-identical to the source raw string. }} and {{ outside a paired {{name}} reference should be left alone.

Actual: every }} is rewritten to >>, regardless of whether it forms a paired variable reference. Bodies containing nested stringified JSON, regex patterns, or simply two adjacent closing braces are silently corrupted.

[repro.json](https://github.com/user-attachments/files/27808839/repro.json)

### Deployment Type

Self-hosted (on-prem deployment)

### Version

v2026.4.0

### PR DESCRIPTION
**Pull request by Sraman01-code**
Fixes #6324

## Summary

This updates Postman variable conversion so only complete paired Postman variable references such as `{{serviceInternalUrl}}` are translated to Hoppscotch variables.

Previously, standalone `}}` sequences inside raw request bodies could be converted to `>>`, corrupting valid JSON.

## Changes

- Replaced independent `{{` and `}}` substitutions with a single matched-pair replacement.
- Added regression coverage for:
  - preserving literal adjacent closing braces in raw JSON request bodies
  - converting valid Postman variables without altering unrelated braces

## Verification

```bash
cd packages/hoppscotch-common
pnpm vitest run src/helpers/__tests__/postman-import.spec.ts

<!-- This is an auto-generated description by cubic. -->
---
## Summary by cubic
Fix Postman import variable conversion to only translate complete, non-empty `{{...}}` tokens into `<<...>>`. This preserves literal braces and empty brace pairs in raw bodies and prevents JSON corruption.

- **Bug Fixes**
  - Use a matched-pair replacement: `/{{\s*([^{}\s]+)\s*}}/g` -> `<<$1>>`.
  - Add regression tests covering literal adjacent `}}` and empty/whitespace-only `{{}}`, while still converting valid Postman variables.

<sup>Written for commit 3264122c0138d0ac903e5a7a415dd673f47741e5. Summary will update on new commits. <a href="https://cubic.dev/pr/hoppscotch/hoppscotch/pull/6328?utm_source=github">Review in cubic</a></sup>

<!-- End of auto-generated description by cubic. -->

### CODE REVIEW

File: packages/hoppscotch-common/src/helpers/__tests__/postman-import.spec.ts [hunk 1]
```diff
@@ -30,7 +30,81 @@ const postmanCollectionWithArrayHeader = JSON.stringify({
...
+    expect(E.isRight(result)).toBe(true)
+
+    if (E.isLeft(result)) {
+      throw new Error("Expected Postman import to succeed")
+    }
+
+    expect(result.right[0].requests[0].body.body).toBe(
+      '{"url":"<<serviceInternalUrl>>","data":{"nested":{"k":"v"}}}'
+    )
+  })
```
[the remark this answers is not in the document]
Comment (Sraman01-code) [reply packages/hoppscotch-common/src/helpers/__tests__/postman-import.spec.ts:136]: Added direct unit coverage for `replacePMVarTemplating`, including valid variable conversion, whitespace trimming, literal brace preservation, multiple variables, and empty/whitespace-only token behavior.

**Comment by Sraman01-code** [comment]
Thanks! The bot review found no issues. Could a maintainer please approve the pending workflow when convenient? Happy to address any review feedback.

**Review by Sraman01-code — COMMENTED** [review]
All Okay
